app/src/androidTest/java/net/aiscope/gdd_app/ui/metadata/SampleImagesAdapterTest.kt

```diff
@@ -0,0 +1,69 @@
+package net.aiscope.gdd_app.ui.metadata
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.File
+import java.io.InputStream
+
+@RunWith(AndroidJUnit4::class)
+class SampleImagesAdapterTest {
+    private val testContext = InstrumentationRegistry.getInstrumentation().context
+    private val targetContext = InstrumentationRegistry.getInstrumentation().targetContext
+    private val cacheDir = targetContext.cacheDir
+    private lateinit var targetFile: File
+
+    @Before
+    fun setup() {
+        targetFile = File(targetContext.filesDir, "test.bmp");
+        targetFile.createNewFile()
+
+        val stream = testContext.assets.open("w3c_home.bmp");
+        stream.toFile(targetFile)
+    }
+
+
+    @Test
+    fun shouldWriteToCache() {
+        val b = runBlocking {
+            decodeSampledBitmapAndCache(targetFile, 20, 20, cacheDir)
+        }
+
+        assertTrue(cacheDir.isDirectory)
+        val cachedFiles = cacheDir.listFiles()
+
+        assertTrue(cachedFiles.any { file -> file.name.equals("test_20x20.bmp", true) })
+        assertTrue(b.height < 50)
+        assertTrue(b.width < 50)
+        assertTrue(b.height > 20)
+        assertTrue(b.width > 20)
+    }
+
+    @Test
+    fun shouldNotWriteToCacheTwice() {

```

Comment:
So apparently the `lastmodified` is in seconds. So the test runs in <1 second and you get the same value on `lastmodified`

I added a Thread.sleep(2000) between the first and second calls, and then it will fail if you do not return the cached file. But that's a bit, ewww... Suggestions?

Reply:
Would it work with less than 2 seconds? :D 
An alternative would be to maybe use a `WatchService` to see if the file has been modified, but not sure if it's worth it. Here is an article on how it would work: https://docs.oracle.com/javase/tutorial/essential/io/notification.html

Reply:
Hahaha well, bigger than 1 second at least. 1200 ms would be acceptable I guess :P 

I don't think the `WatchService` the best idea, I was thinking more along the lines of using Mockito to see if `writeToFile` gets called twice. Not _exactly_ the same test but similar enough I guess.

Reply:
Changed it to `1200ms` for now